protocols/ntree/ntree.go

```diff
@@ -0,0 +1,132 @@
+// Package ntree implements a scheme where a leader (the root node) collects N standard individual signatures
+// from the N witnesses using a tree. As the "naive" scheme where the leader directly sends the message to be signed
+// directly to its children is a special case if ntree (a 1-level tree) this can also be used to measure how the naive
+// approach compares to ntree and CoSi.
+package ntree
+
+import (
+	"fmt"
+	"github.com/dedis/cothority/lib/crypto"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+func init() {
+	// register network messages and protocol
+	network.RegisterMessageType(Message{})
+	network.RegisterMessageType(SignatureReply{})
+}
+
+// Protocol implements the sda.ProtocolInstance interface
+type Protocol struct {
+	*sda.Node
+	// the message we want to sign (and the root node propagates)
+	message []byte
+	// signature of this particular participant:
+	signature crypto.SchnorrSig
+}
+
+func NewProtocol(node *sda.Node) (*Protocol, error) {
+	p := &Protocol{
+		Node: node,
+	}
+	err := p.RegisterHandler(p.HandleSignRequest)
+	if err != nil {
+		return nil, fmt.Errorf("Couldn't register handler %v", err)
+	}
+	err = p.RegisterHandler(p.HandleSignReply)
+	if err != nil {
+		return nil, fmt.Errorf("Couldn't register handler %v", err)
+	}
+	return p, nil
+}
+
+func (p *Protocol) Start() error {
+	if p.IsRoot() {
+		dbg.Lvl3("Starting ntree/naive")
+		return p.HandleSignRequest(structMessage{p.TreeNode(), Message{p.message}})
+	} else {
+		return fmt.Errorf("Called Start() on non-root ProtocolInstance")
+	}
+}
+
+func (p *Protocol) HandleSignRequest(msg structMessage) error {
```

Comment:
About HandleSignRequest + HandleSignReply
* No need for them to be exported. handleSign* is enough I think.
* A small comment would be most welcomed. Same for handleSignReply.